src/core/fft_helpers.cpp

```diff
@@ -28,214 +29,251 @@
 #include <cmath>
 #include "lmms_constants.h"
 
-/* returns biggest value from abs_spectrum[spec_size] array
-
-   returns -1 on error
-*/
-float maximum(float *abs_spectrum, unsigned int spec_size)
-{
-	float maxi=0;
+/* Returns biggest value from abs_spectrum[spec_size] array.
+ *
+ *	returns -1 on error
+ *	returns maximum
+ */
+float maximum(float *abs_spectrum, unsigned int spec_size) {
+	float maxi = 0;
 	unsigned int i;
 
-	if ( abs_spectrum==NULL )
-		return -1;
+	if (abs_spectrum == NULL) {return -1;}
+	if (spec_size <= 0) {return -1;}
 
-	if (spec_size<=0)
-		return -1;
-
-	for ( i=0; i<spec_size; i++ )
-	{
-		if ( abs_spectrum[i]>maxi )
-			maxi=abs_spectrum[i];
+	for (i = 0; i < spec_size; i++) {
```

Comment:
Umm, the coding conventions actually say nothing of that sort -- I checked this specifically at least twice before I started coding.
There is a rule that says every if / for / while / ... statement must use _explicit blocking_ (that is, curly braces cannot be omitted) and that it specifically applies to one-liners as well. But it does not specify where the block should start. At best it hints at starting at a new line in the example, but, well, the example had to use _some_ version of the two.
Around the same time, I saw a discussion on Discord (techtalk, probably) on the same topic, and I interpreted the result as "both styles are OK", so I just stuck with the one I'm used to..

Reply:
The convention throughout the entire codebase is to put the opening and closing braces on their own lines. Since the entire goal of the coding conventions page is to codify conventions (causing consistency), the correct choice is clear.

I'm so sorry about the above, the alliteration started adding up and I couldn't help but go along with it. In plain and readable language: Johannes is correct, you should place your curly-braces on separate lines. I agree that it's awkward, but it's how we do things (for now, at least).

Reply:
Sorry @he29-net , the conventions were not clear here, but as @Spekular says, the code should look consistent in the end. It was like an implicit rule. I updated the [coding conventions](https://github.com/LMMS/lmms/wiki/Coding-conventions) page now.

Reply:
OK, thanks for clarification and the conventions update, I will move the opening brackets to separate lines.